Alex Setyawan iP

alex-setyawan opened the pull request:
### AWEX

> "When in doubt, look intelligent." - Garrison Keillor

And that is exactly what using AWEX allows you to do! Why record tasks using a notepad or a calendar, when you can painstakingly download a rudimentary project from online, that provides the exact same functionality?

Notepad? Boring. Overused. Plain.

Chatbot? Cool. Fancy. [_Sexy_.](https://en.wikipedia.org/wiki/Sexy)

AWEX (**A** **W**onderful **EX**perience) revolutionises the way we construct our to-do lists, being

- text-based
- easy to learn
- entertaining to watch,

and letting you,

- [x] create tasks
- [x] delete tasks
- [x] find tasks

unlike ANY other productivity app! 😱 

All you need to do is:

1. Wait for me to finish it.
2. Explore various better alternatives in the meantime.

Want to see how we make our tasks? Here's a sneak peek of our `Task` class!

```
public class Task {
    private String what;
    private String status;
    private String type;
    .
    .
    .
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

alex-setyawan committed:
started level 0

alex-setyawan committed:
added level 1

alex-setyawan committed:
added level 2

alex-setyawan committed:
added level 3

alex-setyawan committed:
added level 4

alex-setyawan committed:
added text UI testing

alex-setyawan committed:
added level 5

alex-setyawan committed:
added level 6

alex-setyawan committed:
add JavaDocs to Task.java

alex-setyawan committed:
Add JavaDocs to TodoTask.java

alvislowyh commented:
Add code descriptions and change method names such as.. message() to displayMessage?

Jajared commented on src/main/java/Awex.java:
```diff
@@ -0,0 +1,130 @@
+import java.util.*;
+
+public class Awex {
+    public static void message() {
```
Good abstraction of the message to display the list of commands available. Perhaps a better and more specific variable name could be used here to describe this function? For example you could rename it to: `listAllCommands()`

Jajared commented on src/main/java/Awex.java:
```diff
@@ -0,0 +1,130 @@
+
+    public static void main(String[] args) {
+        System.out.println("Hello! I'm AWEX!\nWhat can I do for you?");
+        LinkedList<Task> list = new LinkedList<>();
```
Any reason why a LinkedList is used instead of an ArrayList? Indeed a LinkedList works similar to an ArrayList, although their use cases may vary depending on the scenario. Here is some research that I did: [ArrayList vs LinkedList](https://www.w3schools.com/java/java_linkedlist.asp)

Jajared commented on src/main/java/Awex.java:
```diff
@@ -0,0 +1,130 @@
+import java.util.*;
```
Perhaps you could try to avoid the use of wildcard imports here as it could lead to import conflicts.

Jajared commented on src/main/java/Awex.java:
```diff
@@ -0,0 +1,130 @@
+        Scanner sc = new Scanner(System.in);
+        String next = sc.nextLine();
+        String[] arr = next.split(" ");
+        while (!next.equals("bye")) {
```
Good use of if-else statements to check and parse inputs! However, perhaps you could abstract out this method and also do consider using switch-case statements to improve performance.

Jajared commented on src/main/java/DeadlineTask.java:
```diff
@@ -0,0 +1,12 @@
+public class DeadlineTask extends Task {
```
Good use of inheritance!

Jajared reviewed (commented):
Overall good use of OOP principles and general adherence to coding standards. Some improvements and refactoring can be done to clean up the code and improve readability. Keep up the good work!

alex-setyawan committed:
Clean up code, add more abstractions and JavaDocs

alex-setyawan committed:
Add factory methods for DeadlineTask.java and EventTask.java

alex-setyawan committed:
Add Level 7

alex-setyawan committed:
Remove JavaDocs from master branch

alex-setyawan committed:
Deconflicted TodoTask.java

alex-setyawan committed:
Add various changes

alex-setyawan committed:
Add Level-8

alex-setyawan committed:
Add JavaDocs to Task.java

alex-setyawan committed:
Merge branch 'branch-Level-8'

alex-setyawan committed:
Add Parser, Storage, TaskList and Ui classes

alex-setyawan committed:
Add A-MoreOOP

alex-setyawan committed:
Merge branch 'branch-A-MoreOOP'

alex-setyawan committed:
Add A-Packages

alex-setyawan committed:
Change runtest.bat

alex-setyawan committed:
Change filepaths to include packages

alex-setyawan committed:
Merge branch 'add-gradle-support'

alex-setyawan committed:
Add A-JUnit

alex-setyawan committed:
Merge branch 'branch-A-JUnit'

alex-setyawan committed:
Add A-Jar

alex-setyawan committed:
Add some JavaDocs for A-JavaDoc

alex-setyawan committed:
Merge branch 'branch-A-JavaDoc'

alex-setyawan committed:
Add A-CodingStandard

alex-setyawan committed:
Merge branch 'branch-A-CodingStandard'

alex-setyawan committed:
Add Level-9

alex-setyawan committed:
Merge branch 'branch-Level-9'

alex-setyawan committed:
Fix bug in TaskList.java

alex-setyawan committed:
Add first attempt of Level-10

alex-setyawan committed:
Add second attempt of Level-10

alex-setyawan committed:
Merge branch 'branch-Level-10'

alex-setyawan committed:
Add A-Assertions

alex-setyawan committed:
Add A-CodeQuality

Currently, all logic of command identification is done in Awex.getResponse() method

This leads to getResponse being long and confusing, leading to potential bugs

Abstracting each if-else block in the current getResponse method helps for better
readability, as well as ease in narrowing down sources of bugs.

alex-setyawan committed:
Merge pull request #2 from alex-setyawan/branch-A-Assertions

Add A-Assertions

alex-setyawan committed:
Merge branch 'master' of https://github.com/alex-setyawan/ip into branch-A-CodeQuality

alex-setyawan committed:
Add some JavaDocs

alex-setyawan committed:
Merge branch 'branch-A-CodeQuality'

alex-setyawan committed:
Add C-Help extension

Currently, users always get the instruction message if they type erroneous inputs,
regardless if they know how to rectify them.

As the instruction message is long, it runs the risk of clogging the conversation space.

So a message directing the user to type "help" for full instructions is made,
so the user can get advice only when he/she needs it.

alex-setyawan committed:
Merge branch 'branch-BCD-Extension'

alex-setyawan committed:
Add A-UserGuide

alex-setyawan committed:
Add more JavaDocs

alex-setyawan committed:
Change mainClass in build.gradle
